sdk/core/azure-core-tracing-opentelemetry/test/ut/service_support_test.cpp

```diff
@@ -425,33 +427,30 @@ TEST_F(OpenTelemetryServiceTests, NestSpans)
     auto provider(
         Azure::Core::Tracing::OpenTelemetry::OpenTelemetryProvider::Create(tracerProvider));
 
-    Azure::Core::Context::ApplicationContext.SetTracerProvider(provider);
-
     Azure::Core::Http::Request outerRequest(
         HttpMethod::Post, Azure::Core::Url("https://www.microsoft.com"));
     Azure::Core::Http::Request innerRequest(
         HttpMethod::Post, Azure::Core::Url("https://www.microsoft.com"));
     {
       Azure::Core::_internal::ClientOptions clientOptions;
       clientOptions.Telemetry.ApplicationId = "MyApplication";
+      clientOptions.Telemetry.TracingProvider = provider;
 
       Azure::Core::Tracing::_internal::TracingContextFactory serviceTrace(
-          clientOptions, "my-service", "1.0beta-2");
+          clientOptions, "my.service", "1.0beta-2");
 
-      Azure::Core::Context parentContext;
       Azure::Core::Tracing::_internal::CreateSpanOptions createOptions;
       createOptions.Kind = Azure::Core::Tracing::_internal::SpanKind::Client;
-      auto contextAndSpan
-          = serviceTrace.CreateTracingContext("My API", createOptions, parentContext);
+      auto contextAndSpan = serviceTrace.CreateTracingContext("My API", createOptions, {});
```

Comment:
These are arbitrary strings, the only requirement is that the input strings match the string values which are being checked. 

If this was sample code, it might make a difference. But in this case it's not.

The samples in the DistributedTracing.md use the correctly structured names, as do the versions in `TemplateClient` and the collective `AttestationClient` classes.

Reply:
Btw, the service example (at around line 582 or so) is an example of a service client. It uses the correct names for the service methods.